baybe/surrogates/base.py

```diff
@@ -147,10 +178,15 @@ def fit(self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor) -> Non
                 "Continuous search spaces are currently only supported by GPs."
             )
 
-        # Validate and prepare the training data
-        train_x = _prepare_inputs(train_x)
-        train_y = _prepare_targets(train_y)
+        # Store context-specific transformations
+        self._input_transform = lambda x: searchspace.transform(x, allow_missing=True)
```

Comment:
Technically, you set these functions before calling `_fit`, so the description that you give in the docstrings regarding these functions is not entirely correct. Very picky, I know, but still ;)
Also, if these functions are just taking the `transform` functions from different objects involved, why do we hae them as fields here at all? Wouldn't it be sufficient to just call these functions here without storing them?

Reply:
I don't understand your first point, can you elaborate? The attributes `self._input_transform` and `self._target_transform` are `None` before `fit` is called, that's what the docstring tries to convey.

Regarding second part: The logic is that the surrogates are now responsible for transforming data (currently the idea is that this will comprise all steps from `exp_rep` to what is finally passed to their internal models, so that can be the pure `exp_rep` to `comp_rep`, but potentially also scaling later – needs to be fleshed out in the next PRs). So outside the surrogate, it's unclear how that transformation will look like. Unfortunately, though, we also need to be able to perform this transformation in `AcquisitionFunction`, that's why we expose the fitted transform via the public `transform_inputs`/`transform_targets` functions , that internally access these fields.

As I said, the final layout is not yet fully decided and will become clearer in the next PRs, but the current idea is to abstract away the specific transformation applied to the outside world

Reply:
@AVHopp, can you comment? Last blocking thread before merge

Reply:
Sorry, missed that. First point can be ignored (that was a not too serious comment ;)) and I am happy with your 2nd explanation.